subsys/bluetooth/host/l2cap.c

```diff
@@ -2411,6 +2411,15 @@ void bt_l2cap_recv(struct bt_conn *conn, struct net_buf *buf)
 		return;
 	}
 
+	/* if packet was not complete this means peer device overflowed our
+	 * RX and channel shall be disconnected
+	 */
+	if (!complete) {
```

Comment:
What about the case where `conn->type == BT_CONN_TYPE_BR`? 

Wouldn't you need to forward the `complete` boolean to that, and add the same handling in `bt_l2cap_br_recv`?

Reply:
TBH I don't know (and I have no way of testing that)

Reply:
I am not too familiar with the BR code either; @jhedberg do you know?

Reply:
this fix is intended to be backported to 2.7 for LE qualification,  do we really need to discuss BR/EDR here? (which is likely unused anyway).  Even if BR/EDR is to be fixed I think it should be separate PR anyway

Reply:
> Even if BR/EDR is to be fixed I think it should be separate PR anyway

You are changing the behavior of BR/EDR as well in this PR as far as I can tell, so I disagree with this statement. 

@jhedberg and @joerchan probably have better input than me on this, so if they both approve, that's fine with me.

Reply:
I see that the code has been changed, so that this is just for dynamic channels. LGTM :)